File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    stack = []
    for i in s:
        if i == "(":
            stack.append(i)
        else:
            if len(stack) == 0:
                answer = False
                break
            else:
                if stack[-1] == "(":
                    stack.pop()
                else:
                    answer = False
                    break
    
    if len(stack) > 0:
        answer = False
    
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s) {
    var answer = true;

    let stack = [];
    
    for (let i of s) {
        if (i === "(") {
            stack.push(i);
        } else if (i === ")") {
            if (stack.length === 0 || stack[stack.length - 1] !== "(") {
                return false;
            }
            stack.pop();
        }
    }
    
    if (stack.length !== 0) {
        answer = false;
    }
    
    return answer;
}